### ISSUE TITLE
Shape mismatch when exporting to .tflite

### ISSUE THREAD

**Issue by DanBmh** [mozilla/DeepSpeech#3123]
I did try to export a French transfer-learning model with different alphabet than the English one, but I get following error: 
`ValueError: Cannot feed value of shape (44,) for Tensor 'layer_6/bias/Initializer/zeros:0', which has shape '(29,)'`

Exporting works well without the error if I export my German model with the same alphabet size as the English model.

<br>

Tested with docker container at commit #188a6f2c1ee53dc79acf8abceaf729b5f9a05e7a (15 days ago)

Using current master doesn't work for the German model anymore, seems the graph code got changed in-between because now it wants another shape:
`ValueError: Cannot feed value of shape (29,) for Tensor 'layer_6/bias/Initializer/zeros:0', which has shape '(30,)'`

<br>

Full stack trace:
```
root@DANIEL-PC:/DeepSpeech# export CHECKPOINT_DIR=checkpoints/fr/ccmtv/
root@DANIEL-PC:/DeepSpeech# 
root@DANIEL-PC:/DeepSpeech# python3 -u /DeepSpeech/DeepSpeech.py --checkpoint_dir ${CHECKPOINT_DIR}     --export_tflite --export_dir ${CHECKPOINT_DIR}
I Exporting the model...
I Loading best validating checkpoint from /DeepSpeech/checkpoints/fr/ccmtv/best_dev-767109
I Loading variable from checkpoint: cudnn_lstm/rnn/multi_rnn_cell/cell_0/cudnn_compatible_lstm_cell/bias
I Loading variable from checkpoint: cudnn_lstm/rnn/multi_rnn_cell/cell_0/cudnn_compatible_lstm_cell/kernel
I Loading variable from checkpoint: layer_1/bias
I Loading variable from checkpoint: layer_1/weights
I Loading variable from checkpoint: layer_2/bias
I Loading variable from checkpoint: layer_2/weights
I Loading variable from checkpoint: layer_3/bias
I Loading variable from checkpoint: layer_3/weights
I Loading variable from checkpoint: layer_5/bias
I Loading variable from checkpoint: layer_5/weights
I Loading variable from checkpoint: layer_6/bias
Traceback (most recent call last):
  File "/DeepSpeech/DeepSpeech.py", line 12, in <module>
    ds_train.run_script()
  File "/DeepSpeech/training/deepspeech_training/train.py", line 955, in run_script
    absl.app.run(main)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/DeepSpeech/training/deepspeech_training/train.py", line 935, in main
    export()
  File "/DeepSpeech/training/deepspeech_training/train.py", line 788, in export
    load_graph_for_evaluation(session)
  File "/DeepSpeech/training/deepspeech_training/util/checkpoints.py", line 146, in load_graph_for_evaluation
    _load_or_init_impl(session, methods, allow_drop_layers=False)
  File "/DeepSpeech/training/deepspeech_training/util/checkpoints.py", line 97, in _load_or_init_impl
    return _load_checkpoint(session, ckpt_path, allow_drop_layers)
  File "/DeepSpeech/training/deepspeech_training/util/checkpoints.py", line 70, in _load_checkpoint
    v.load(ckpt.get_tensor(v.op.name), session=session)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/util/deprecation.py", line 324, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/variables.py", line 1033, in load
    session.run(self.initializer, {self.initializer.inputs[1]: value})
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/client/session.py", line 956, in run
    run_metadata_ptr)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/client/session.py", line 1156, in _run
    (np_val.shape, subfeed_t.name, str(subfeed_t.get_shape())))
ValueError: Cannot feed value of shape (44,) for Tensor 'layer_6/bias/Initializer/zeros:0', which has shape '(29,)'

```

Do you have an idea how to solve it?

**Comment by reuben** [comment]
Did you modify `data/alphabet.txt` in-place? I don't see an `--alphabet` parameter specified. In any case, the error here indicates you're not passing the same `--alphabet` file at export time that you used for training this checkpoint.

**Comment by reuben** [comment]
> Using current master doesn't work for the German model anymore, seems the graph code got changed in-between because now it wants another shape:
> `ValueError: Cannot feed value of shape (29,) for Tensor 'layer_6/bias/Initializer/zeros:0', which has shape '(30,)'`

This might be a regression from a recent refactoring that removed the Python Alphabet implementation in favor of the C++ one.

**Comment by DanBmh** [comment]
> Did you modify `data/alphabet.txt` in-place? I don't see an `--alphabet` parameter specified. In any case, the error here indicates you're not passing the same `--alphabet` file at export time that you used for training this checkpoint.

Thanks, this solved it. I did try it before, but only in the current master container it seems ... (didn't help there)

**Comment by reuben** [comment]
> > Using current master doesn't work for the German model anymore, seems the graph code got changed in-between because now it wants another shape:
> > `ValueError: Cannot feed value of shape (29,) for Tensor 'layer_6/bias/Initializer/zeros:0', which has shape '(30,)'`
> 
> This might be a regression from a recent refactoring that removed the Python Alphabet implementation in favor of the C++ one.

It looks like it is indeed the case, I was able to reproduce it. Should be fixed by #3125.
